app/src/main/java/io/apicurio/registry/ibmcompat/api/impl/ApiServiceImpl.java

```diff
@@ -70,46 +230,34 @@
                   .map(id -> {
                       SchemaListItem item = new SchemaListItem();
                       try {
-                          StoredArtifact artifact = storage.getArtifact(id);
-                          item.setId(id);
-                          item.setEnabled(true);
-                          SchemaVersion version = new SchemaVersion();
-                          version.setId(artifact.getVersion().intValue()); // TODO not safe!
-                          item.setLatest(version);
+                          populateSchemaSummary(id, item);
+                          item.setLatest(getLatestSchemaVersion(id));
                       } catch (ArtifactNotFoundException e) {
-                          // we can have deleted artifact ...
+                          return null;
```

Comment:
Good change in approach - but I would perhaps add back in a comment explaining what's happening.  i.e. that a race condition could cause the artifact to be not found, so we'll return null here and filter out nulls later.